src/__tests__/pick.test.ts

```diff
@@ -0,0 +1,130 @@
+import { plainToClass } from 'class-transformer';
+import { validateSync } from 'class-validator';
+
+import { IsNested, IsString } from '../decorators';
+import { pick } from '../pick';
+
+describe('pick', () => {
+    class Child {
+        @IsString()
+        baz!: string;
+
+        @IsString()
+        qux!: string;
+    }
+
+    class Parent {
+        @IsString()
+        foo!: string;
+
+        @IsString()
+        bar!: string;
+
+        @IsNested({
+            type: Child,
+        })
+        child!: Child;
+    }
+
+    it('picks a single field and does not validate', () => {
+        const ParentFixture = pick(Parent, 'ParentFixture', ['foo']);
+
+        const obj = plainToClass(ParentFixture, {
+        });
+        const errors = validateSync(obj);
+        expect(errors).toHaveLength(1);
+    });
+    it('picks a single field and validates', () => {
+        const ParentFixture = pick(Parent, 'ParentFixture', ['foo']);
+
+        const obj = plainToClass(ParentFixture, {
+            foo: 'foo',
+        });
+        const errors = validateSync(obj);
+        expect(errors).toHaveLength(0);
+    });
+    it('picks zero fields', () => {
+        const ParentFixture = pick(Parent, 'ParentFixture', []);
+
+        const obj = plainToClass(ParentFixture, {
+        });
+        const errors = validateSync(obj);
+        expect(errors).toHaveLength(0);
+    });
+    it('picks child fields and does not validate', () => {
+        const ParentFixture = pick(Parent, 'ParentFixture', ['child']);
+
+        const obj = plainToClass(ParentFixture, {
+        });
+        const errors = validateSync(obj);
+        expect(errors).toHaveLength(1);
+    });
+    it('picks all child fields and validates', () => {
+        const ParentFixture = pick(Parent, 'ParentFixture', ['child']);
+
+        const obj = plainToClass(ParentFixture, {
+            child: {
+                baz: 'baz',
+                qux: 'sux',
+            },
+        });
+        const errors = validateSync(obj);
+        expect(errors).toHaveLength(0);
+    });
+    it('picks one field and one child field and does not validate if field is omitted', () => {
```

Comment:
One common scenario we run into is picking a nested structure where we want to modify the type of the child DTO.

There is a sort of solution right now:

 1. We can use `pick` to pluck the child fields to our heart's content.
 2. We can use `pick` to pluck the parent fields and create a base class.
 3. We can write a minimal subclass of the parent that connects the parent to the picked child

This is not perfect yet.

 - If we don't include `child` in `ParentBase`'s `pick`, it doesn't get validated.
 - If we do include `child`, we get a compiler error if we try to declare `Child` as `ChildFixture` (or `typeof `ChildFixture`)

The validation still works, so this might be ok on the **internal-to-external** edge of system (since nothing will use the `child` type once we encode JSON). But I want to see about fixing this logic.